r/tests/testthat/helper-expectation.R

```diff
@@ -317,3 +317,7 @@ split_vector_as_list <- function(vec) {
   vec2 <- vec[seq(from = min(length(vec), vec_split + 1), to = length(vec), by = 1)]
   list(vec1, vec2)
 }
+
+err_helper <- function(...) (expect_error(...))
```

Comment:
I like the helpers and that they reduce some of the boiler plate. Any reason not to make them simply `err()` and `warn()` or maybe `err_snap()` and `warn_snap()` to make it clear that they are for use with snapshots? 

If the reason you were thinking of avoiding them is to avoid running into testthat's version if they implement them, it [looks like they aren't planning to](https://github.com/r-lib/testthat/issues/1471#issuecomment-962982383)

Reply:
Aaah I didn't see this earlier: https://github.com/apache/arrow/pull/11534#issuecomment-962983868 yeah, avoiding the rlang conflict is probably a good one, how about `_snap()` so it's clear where to use them? `_helper()` is a _little bit_ too generic to me

Reply:
I have no problem with `err()`, but there already is a `warn()` function in {rlang} so I was a bit concerned with that. I like `err_snap()` and `warn_snap()` though, so it looks like that would be the way to go.

Reply:
Yeah, `err_snap()` and `warn_snap()` are nice to me, I would be curious if @thisisnic @romainfrancois or @paleolimbot have any thoughts about these helper names while we still only have a small number to find/replace.

Reply:
More verbose, but how about `expect_error_snapshot()` and `expect_warning_snapshot()`?  Or without the `expect` if that makes more sense (I haven't read this bit of code in-depth yet).  I'm in favour of leaving full words as-is if it's not causing conflicts or overly long names, as it makes easier to guess what's happening.

Reply:
In this specific instance - a wrapper that is very close to the actual function and does not add much in terms of complexity + is not user facing - I'd be in favour of a shorter name (like the initial `err()`). In order to prevent conflicts with {rlang} I think adding the __snap_ prefix should do. So my vote goes for `err_snap()` and `warn_snap()` and maybe, if needed a `mess_snap()`.

Reply:
`expect_snapshot(..., error = TRUE)` fails to capture the "ancestry" of the call. See more [here](https://github.com/tidyverse/dplyr/issues/6044).

Reply:
@paleolimbot one thing that really helped me was `testthat::snapshot_review()`. 

Also the `(expect_snapshot(` + `expect_error()` pattern has the benefit of explicitly expecting an error from a specific expression, as opposed to merely allow errors in one of the expressions in `expect_snapshot(error = TRUE)`, and so have the test fail otherwise.